storage/kv_store.py

```diff
@@ -75,5 +75,8 @@ def _has_key(self, key):
         except KeyError:
             return False
 
+    def get_equal_or_prev(self, key):
+        pass
```

Comment:
I think it's better to either make it @abstractmethod or raise NotImplementedException by default